server/src/main/java/org/opensearch/search/query/SearchEngineResultConversionUtils.java

```diff
@@ -83,9 +83,10 @@ public static Tuple<List<InternalAggregation>, Long> extractSubAggsAndDocCount(A
                 if (aggregator instanceof ValueCountAggregator) {
                     docCount = ((InternalValueCount) subAgg).getValue();
                 }
-//                if (aggregator instanceof CardinalityAggregator) {
-//                    docCount = ((InternalCardinality) subAgg).getValue();
-//                }
+                // This is required for the cases when we group by distinct count by any string term
```

Comment:
We should figure out the reason for `agg_for_doc_count` not getting added. 
IMO Using approximate count as `doc_count` is incorrect.